java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/analyser/deploy/DeployCommand.java

```diff
@@ -188,6 +192,12 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
                     FileUtil.deleteRecursive(i);
                 }
 
+                //update the DB with contaminant information
+                Log.infof("Contaminants: %s", contaminatedPaths);
+                Log.infof("Contaminated GAVS: %s", contaminatedGavs);
+                for (var i : contaminatedGavs.entrySet()) {
+                    gavs.removeAll(i.getValue());
+                }
```

Comment:
Moved these up - I was trying to ensure that the `No content to deploy found in deploy directory` could be thrown.

Reply:
Although in hindsight this breaks `minikube sample-report ` as it leads to 
```
2023-10-10 10:30:26,882 ERROR [com.red.hac.con.ana.dep.DeployCommand$1] (main) io/github/stuartwdouglas/hacbs-test/shaded/shaded-jdk11/1.9/shaded-jdk11-1.9.jar was contaminated by io.github.stuartwdouglas.hacbs-test.simple:simple-jdk8:1.2.4 from central
2023-10-10 10:30:27,162 ERROR [com.red.hac.con.ana.dep.DeployCommand] (main) Removing /workspace/source/artifacts/io/github/stuartwdouglas/hacbs-test/shaded/shaded-jdk11/1.9 as it is contaminated
2023-10-10 10:30:27,166 INFO  [com.red.hac.con.ana.dep.DeployCommand] (main) Contaminants: {io/github/stuartwdouglas/hacbs-test/shaded/shaded-jdk11/1.9=[io.github.stuartwdouglas.hacbs-test.simple:simple-jdk8:1.2.4]}
2023-10-10 10:30:27,172 INFO  [com.red.hac.con.ana.dep.DeployCommand] (main) Contaminated GAVS: {io.github.stuartwdouglas.hacbs-test.simple:simple-jdk8:1.2.4=[io.github.stuartwdouglas.hacbs-test.shaded:shaded-jdk11:1.9]}
2023-10-10 10:30:27,173 INFO  [com.red.hac.con.ana.dep.DeployCommand] (main) GAVs to deploy: []
2023-10-10 10:30:27,173 ERROR [com.red.hac.con.ana.dep.DeployCommand] (main) No content to deploy found in deploy directory
```

Reply:
It needs to go back. "No content to deploy found in deploy directory" means the build failed to produce any deployable output. If everything is contaminated we still need the task to succeed so we can resolve the contamination.